### PR TITLE
tests for method drawDiamond()

### PR DESCRIPTION
**Pull request by Sofasmile**

### CODE REVIEW

File: src/main/java/core/basesyntax/Diamond.java [hunk 1]
```diff
@@ -0,0 +1,21 @@
+package core.basesyntax;
+
+import java.io.PrintStream;
+
+public class Diamond {
+
+    /**
+     * Нарисовать ромб с звёздочек заданной величины.
```
Comment (roman-mate) [inline src/main/java/core/basesyntax/Diamond.java]:
```suggestion
     * Нарисовать ромб со звёздочек заданной величины.
```

File: src/main/java/core/basesyntax/Diamond.java [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+     *
+     * Пример: величина 3, значит максимальное кол-во звёзд в ряду будет.
```
Comment (roman-mate) [inline src/main/java/core/basesyntax/Diamond.java]: this sentence is not complete.

File: src/test/java/core/basesyntax/DiamondTest.java [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+import org.junit.Test;
+
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+
+public class DiamondTest {
+    private static final String ONE_STAR = "* \n";
+
+    private static final String EVEN_NUMBER_OF_STAR =
+                      "   * \n"
```
Comment (roman-mate) [inline src/test/java/core/basesyntax/DiamondTest.java]: do we want this extra space in the end? for the test case to be more reliable we can trim spaces in the end.

File: src/test/java/core/basesyntax/DiamondTest.java [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+                    + "* * * * * \n"
+                    + " * * * * \n"
+                    + "  * * * \n"
+                    + "   * * \n"
+                    + "    * \n";
+
+    @Test
+    public void oneStarScenario() {
+        try {
+            StringStream actualPrintStream = new StringStream(new FileOutputStream("file"));
```
Comment (roman-mate) [inline src/test/java/core/basesyntax/DiamondTest.java]: new FileOutputStream("file") object is a dummy object which is not really used. Let's update StringStream to have empty parameters constructor and use empty contructor here?

File: src/test/java/core/basesyntax/DiamondTest.java [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+                    + "  * * * \n"
+                    + " * * * * \n"
+                    + "* * * * * \n"
+                    + " * * * * \n"
+                    + "  * * * \n"
+                    + "   * * \n"
+                    + "    * \n";
+
+    @Test
+    public void oneStarScenario() {
```
Comment (roman-mate) [inline src/test/java/core/basesyntax/DiamondTest.java]: https://testing.googleblog.com/2014/10/testing-on-toilet-writing-descriptive.html

shouldPrintOneStartDiamonCorrectly.

File: src/test/java/core/basesyntax/DiamondTest.java [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+        try {
+            StringStream actualPrintStream = new StringStream(new FileOutputStream("file"));
+            Diamond.drawDiamond(1, actualPrintStream);
+
+            Assert.assertEquals("Test failed with parameter: maxSize = " + 1,
+                    ONE_STAR,
+                    actualPrintStream.getPrintString());
+        } catch (FileNotFoundException e) {
```
Comment (roman-mate) [inline src/test/java/core/basesyntax/DiamondTest.java]: If test fails with exception in report it'll be shown as passed. We should never ignore exception.

File: src/main/java/core/basesyntax/Diamond.java (hunk 2 above)
Comment (Sofasmile) [reply src/main/java/core/basesyntax/Diamond.java]: done

File: src/main/java/core/basesyntax/Diamond.java (hunk 1 above)
Comment (Sofasmile) [reply src/main/java/core/basesyntax/Diamond.java]: done

File: src/test/java/core/basesyntax/DiamondTest.java (hunk 3 above)
Comment (Sofasmile) [reply src/test/java/core/basesyntax/DiamondTest.java]: done

File: src/test/java/core/basesyntax/DiamondTest.java [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+                    + "  * * *\n"
+                    + " * * * *\n"
+                    + "* * * * *\n"
+                    + " * * * *\n"
+                    + "  * * *\n"
+                    + "   * *\n"
+                    + "    *";
+
+    @Test
+    public void shouldPrintOneStartDiamondCorrectly() {
```
Comment (roman-mate) [inline src/test/java/core/basesyntax/DiamondTest.java]:
```suggestion
    public void shouldPrintOneStarDiamondCorrectly() {
```

File: src/main/java/core/basesyntax/StringStream.java [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+import java.io.PrintStream;
+
+public class StringStream extends PrintStream {
+    private String printString = "";
+
+    public StringStream() throws FileNotFoundException {
+        super(new FileOutputStream("file"));
```
Comment (roman-mate) [inline src/main/java/core/basesyntax/StringStream.java]: Let's wrap code that creates new FileOutputStream into try catch. Catch exception and rethrow RuntimeException. Than you won't have to handle all these useless FileNotFoundExceptions.

File: src/test/java/core/basesyntax/DiamondTest.java (hunk 7 above)
Comment (Sofasmile) [reply src/test/java/core/basesyntax/DiamondTest.java]: done

File: src/main/java/core/basesyntax/StringStream.java (hunk 8 above)
Comment (Sofasmile) [reply src/main/java/core/basesyntax/StringStream.java]: done

File: src/test/java/core/basesyntax/DiamondTest.java [hunk 9]
```diff
@@ -0,0 +1,65 @@
...
+        Assert.assertEquals("Test failed with parameter: maxSize = " + 1,
+                ONE_STAR,
+                actualResult);
+    }
+
+    @Test
+    public void shouldPrintEvenStarDiamondCorrectly() {
+        StringStream actualPrintStream = new StringStream();
+        Diamond.drawDiamond(4, actualPrintStream);
+        trimStar(actualPrintStream, EVEN_NUMBER_OF_STAR, 4);
```
Comment (roman-mate) [inline src/test/java/core/basesyntax/DiamondTest.java]:
```suggestion
        assertTrimmedDiamondsEqual(actualPrintStream, EVEN_NUMBER_OF_STAR, 4);
```

**Pull request merged**